yb-voyager/src/datafile/datafile.go

```diff
@@ -25,8 +26,10 @@ var reCopy = regexp.MustCompile(`(?i)COPY .* FROM STDIN;`)
 
 func OpenDataFile(filePath string, descriptor *Descriptor) (DataFile, error) {
 	switch descriptor.FileFormat {
-	case CSV, TEXT:
+	case CSV:
 		return openCsvDataFile(filePath, descriptor)
+	case TEXT, CSV_NO_NEWLINE:
```

Comment:
CSV_NO_NEWLINE introduces unnecessary special casing. All you need is on env variable `CSV_NO_NEWLINE=y`. Then, in the `OpenDataFile()` you can first check if `FileFormat == CSV && os.GetEnv("CSV_NO_NEWLINE") == "y"` and, if true, open the file as a TEXT file. Else (i.e. the env var is not set), continue with the default switch case.